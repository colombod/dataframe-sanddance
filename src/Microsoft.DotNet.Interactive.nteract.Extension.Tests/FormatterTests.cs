using System;
using System.Diagnostics;
using System.Linq;
using Assent;
using FluentAssertions;
using FluentAssertions.Execution;
using HtmlAgilityPack;
using Microsoft.DotNet.Interactive.Formatting;
using Xunit;

namespace Microsoft.DotNet.Interactive.nteract.Extension.Tests
{
    public class FormatterTests : IDisposable
    {
        private readonly Configuration _configuration;

        public FormatterTests()
        {
            DataExplorerExtensions.RegisterFormatters();
            _configuration = new Configuration()
                .SetInteractive(Debugger.IsAttached);

        }

        [Fact]
        public void can_generate_tabular_json()
        {
            var data = new[]
            {
                new {Name = "Q", IsValid =false, Cost=10.0},
                new {Name = "U", IsValid =false, Cost=5.0},
                new {Name = "E", IsValid =true, Cost=10.0},
                new {Name = "S", IsValid =false, Cost=10.0},
                new {Name = "T", IsValid =false, Cost=10.0}
            };

            var formattedData = data.ToTabularData();

            
            this.Assent(formattedData.ToString(), _configuration);
        }

        [Fact]
        public void can_format_data()
        {
            var data = new[]
            {
                new {Name = "Q", IsValid =false, Cost=10.0},
                new {Name = "U", IsValid =false, Cost=5.0},
                new {Name = "E", IsValid =true, Cost=10.0},
                new {Name = "S", IsValid =false, Cost=10.0},
                new {Name = "T", IsValid =false, Cost=10.0}
            };

            var formattedData = data.ToDisplayString(TableFormatter.MimeType);


            this.Assent(formattedData, _configuration);
        }

        [Fact]
        public void can_generate_widget()
        {
            var data = new[]
            {
                new {Name = "Q", IsValid =false, Cost=10.0},
                new {Name = "U", IsValid =false, Cost=5.0},
                new {Name = "E", IsValid =true, Cost=10.0},
                new {Name = "S", IsValid =false, Cost=10.0},
                new {Name = "T", IsValid =false, Cost=10.0}
            };

            var explorer = data.Explore();

            var formatted = explorer.ToDisplayString(HtmlFormatter.MimeType);

            var htmlDoc = new HtmlDocument();
            htmlDoc.LoadHtml(formatted);

            var scripts = htmlDoc.DocumentNode.SelectNodes("//div/script").Single();

            using var _ = new AssertionScope();
            scripts.InnerText.Should().Contain(data.ToTabularData().ToString());

            scripts.InnerText.Should().Contain("dotnetInteractiveExtensionsRequire(['dotnet-interactive-extensions/nteract/resources/lib'],");

        }

        public void Dispose()
        {
            Formatter.ResetToDefault();
        }
    }
}
