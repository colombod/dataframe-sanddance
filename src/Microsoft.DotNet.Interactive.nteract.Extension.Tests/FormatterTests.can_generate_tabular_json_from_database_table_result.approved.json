{
  "schema": {
    "primaryKey": [],
    "fields": [
      {
        "name": "ProductID",
        "type": "integer"
      },
      {
        "name": "Name",
        "type": "string"
      },
      {
        "name": "ProductNumber",
        "type": "string"
      },
      {
        "name": "MakeFlag",
        "type": "boolean"
      },
      {
        "name": "FinishedGoodsFlag",
        "type": "boolean"
      },
      {
        "name": "Color",
        "type": "string"
      },
      {
        "name": "SafetyStockLevel",
        "type": "string"
      },
      {
        "name": "ReorderPoint",
        "type": "string"
      },
      {
        "name": "StandardCost",
        "type": "string"
      },
      {
        "name": "ListPrice",
        "type": "string"
      },
      {
        "name": "Size",
        "type": "string"
      },
      {
        "name": "SizeUnitMeasureCode",
        "type": "string"
      },
      {
        "name": "WeightUnitMeasureCode",
        "type": "string"
      },
      {
        "name": "Weight",
        "type": "string"
      },
      {
        "name": "DaysToManufacture",
        "type": "integer"
      },
      {
        "name": "ProductLine",
        "type": "string"
      },
      {
        "name": "Class",
        "type": "string"
      },
      {
        "name": "Style",
        "type": "string"
      },
      {
        "name": "ProductSubcategoryID",
        "type": "string"
      },
      {
        "name": "ProductModelID",
        "type": "string"
      },
      {
        "name": "SellStartDate",
        "type": "date"
      },
      {
        "name": "SellEndDate",
        "type": "string"
      },
      {
        "name": "DiscontinuedDate",
        "type": "string"
      },
      {
        "name": "rowguid",
        "type": "string"
      },
      {
        "name": "ModifiedDate",
        "type": "date"
      }
    ]
  },
  "data": [
    {
      "ProductID": 1,
      "Name": "Adjustable Race",
      "ProductNumber": "AR-5381",
      "MakeFlag": false,
      "FinishedGoodsFlag": false,
      "Color": null,
      "SafetyStockLevel": 1000,
      "ReorderPoint": 750,
      "StandardCost": 0.0000,
      "ListPrice": 0.0000,
      "Size": null,
      "SizeUnitMeasureCode": null,
      "WeightUnitMeasureCode": null,
      "Weight": null,
      "DaysToManufacture": 0,
      "ProductLine": null,
      "Class": null,
      "Style": null,
      "ProductSubcategoryID": null,
      "ProductModelID": null,
      "SellStartDate": "2008-04-30T00:00:00",
      "SellEndDate": null,
      "DiscontinuedDate": null,
      "rowguid": "694215b7-08f7-4c0d-acb1-d734ba44c0c8",
      "ModifiedDate": "2014-02-08T10:01:36.827"
    },
    {
      "ProductID": 2,
      "Name": "Bearing Ball",
      "ProductNumber": "BA-8327",
      "MakeFlag": false,
      "FinishedGoodsFlag": false,
      "Color": null,
      "SafetyStockLevel": 1000,
      "ReorderPoint": 750,
      "StandardCost": 0.0000,
      "ListPrice": 0.0000,
      "Size": null,
      "SizeUnitMeasureCode": null,
      "WeightUnitMeasureCode": null,
      "Weight": null,
      "DaysToManufacture": 0,
      "ProductLine": null,
      "Class": null,
      "Style": null,
      "ProductSubcategoryID": null,
      "ProductModelID": null,
      "SellStartDate": "2008-04-30T00:00:00",
      "SellEndDate": null,
      "DiscontinuedDate": null,
      "rowguid": "58ae3c20-4f3a-4749-a7d4-d568806cc537",
      "ModifiedDate": "2014-02-08T10:01:36.827"
    },
    {
      "ProductID": 3,
      "Name": "BB Ball Bearing",
      "ProductNumber": "BE-2349",
      "MakeFlag": true,
      "FinishedGoodsFlag": false,
      "Color": null,
      "SafetyStockLevel": 800,
      "ReorderPoint": 600,
      "StandardCost": 0.0000,
      "ListPrice": 0.0000,
      "Size": null,
      "SizeUnitMeasureCode": null,
      "WeightUnitMeasureCode": null,
      "Weight": null,
      "DaysToManufacture": 1,
      "ProductLine": null,
      "Class": null,
      "Style": null,
      "ProductSubcategoryID": null,
      "ProductModelID": null,
      "SellStartDate": "2008-04-30T00:00:00",
      "SellEndDate": null,
      "DiscontinuedDate": null,
      "rowguid": "9c21aed2-5bfa-4f18-bcb8-f11638dc2e4e",
      "ModifiedDate": "2014-02-08T10:01:36.827"
    }
  ]
}